lxd/metrics/api_rates.go

```diff
@@ -0,0 +1,102 @@
+package metrics
+
+import (
+	"net/url"
+	"sync/atomic"
+
+	"github.com/canonical/lxd/shared/api"
+	"github.com/canonical/lxd/shared/entity"
+)
+
+// RequestResult represents a completed request status category.
+type RequestResult string
```

Comment:
As we are storing this into completedRequests as the key, shall we make this an int type and add a `String()` function to convert to its presentation name?